### PR TITLE
Pull correct rpm depending on the os name

### PR DESCRIPTION
**Pull request by amdkila**
This PR ensures that sles jobs will pull sles-labelled rpms, and centOS7 jobs will pull centos-labelled rpms. Can easily be modified as we package other distros

### CODE REVIEW

File: vars/auxiliary.groovy [hunk 1]
```diff
@@ -89,6 +89,8 @@ String getLibrary( String projectName, String label, String branchName, boolean
     String permissions = ''
     String packageType
     String packageManager
+    String[] distro = ['ubuntu', 'centos', 'sles']
```
Comment (saadrahim) [inline vars/auxiliary.groovy]: centos7 not centos, we will have centos8 soon. Ubuntu we will likely have just one but we should consider 16/18. SLES we should specify 15.

Comment (saadrahim) [reply vars/auxiliary.groovy]: Write the code for el8, might as well get that done.

Comment (saadrahim) [reply vars/auxiliary.groovy]: Leave it untested ofcourse

Comment (saadrahim) [reply vars/auxiliary.groovy]: as we have no el8 support yet

File: vars/auxiliary.groovy [hunk 2]
```diff
@@ -89,6 +89,9 @@ String getLibrary( String projectName, String label, String branchName, boolean
     String permissions = ''
     String packageType
     String packageManager
+    String[] distro = ['ubuntu16', 'ubuntu18', 'centos7', 'centos8', 'sles15']
```
Comment (saadrahim) [inline vars/auxiliary.groovy]: http://docs.groovy-lang.org/latest/html/documentation/working-with-collections.html#Collections-Maps

**Comment by amdkila** [comment]
Tested this in rocSOLVER, looks like it works

**Pull request merged**